lib/testing/src/lib/process-services/actions/user-filters.util.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from '../../core/utils/logger';
+import { ApiService } from '../../core/actions/api.service';
+import { UserFilterOrderRepresentation, UserTaskFilterRepresentation } from '@alfresco/js-api';
+
+export class UserFiltersUtil {
+
+    apiService: ApiService;
+
+    constructor(apiService: ApiService) {
+        this.apiService = apiService;
+    }
+
+    async createATaskFilter(newTaskFilterName: string, sortType?: string,  stateType?: string, assignmentType?: string, iconName?: string, appId?: number): Promise<any> {
+        try {
+            return this.apiService.getInstance().activiti.userFiltersApi.createUserTaskFilter(new UserTaskFilterRepresentation(
+                {appId: appId, name: newTaskFilterName, icon: iconName, filter: {sort: sortType, state: stateType, assignment: assignmentType}}));
+        } catch (error) {
+            Logger.error('Create Task Filter - Service error, Response: ', JSON.parse(JSON.stringify(error)));
```

Comment:
In these cases, don't we want rethrow the error, beside displaying them? Like this, even if an error occurs, we just show a message, but the program's execution will continue. Is it exactly what we want here?

Reply:
Indeed, the `catch` block will never be executed in case of the errors in the async `createUserTaskFilter`. I suggest not using try/catch here at all, as it's a test and we will see the error + stack trace anyway

Reply:
@popovicsandras, @DenysVuika thanks for the comments. Regarding the `catch's logger.error`... I was writing this in the same idea of what we already have (eg. [lib/process-services/src/lib/process-list/services/process.service.ts](https://github.com/Alfresco/alfresco-ng2-components/blob/develop/lib/process-services/src/lib/process-list/services/process.service.ts)) and I was keeping it like this because I was wondering that if not, we will stop the execution of the tests and the entire class will fail, right? I'm not sure if I'm actually right.
Currently, in a class we have multiple tests, most of them with different preconditions [this is available at least for all the tests from adw->Process Services extension]. If one precondition (required for just one test) will fail, currently the tests will keep running. From my point of view, we could keep using this approach until we will find a way to avoid making an entire class fail because of one precondition needed by a single test. 🤔  

@DenysVuika I'm not sure if I've understood correctly the concern, but `createUserTaskFilter` is an API used directly from js-api package ([node_modules/@alfresco/js-api/src/api/activiti-rest-api/api/userFilters.api.ts](https://github.com/Alfresco/alfresco-js-api/blob/develop/src/api/activiti-rest-api/api/userFilters.api.ts#L73)). If something happens there, shouldn't be caught in `async createATaskFilter` `catch's Logger.error`? 🤔